example-mouse-picker/src/ofApp.cpp

```diff
@@ -41,16 +41,34 @@ void ofApp::draw(){
     unsigned int indexIntersectedPrimitive = 0;
     float distanceToClosestIntersection = numeric_limits<float>::max();
 
+    auto & mouseRay = mousepicker.getRay();
+
     // iterates through all the primitives and find the closest one.
     for (unsigned int i = 0; i<icospheres.size(); i++) {
         auto & ico = icospheres.at(i);
         ico.draw();
 
-        bool intersects = mousepicker.getRay().intersectsPrimitive(ico,  baricentricCoordinates, surfaceNormal);
-        if (intersects && (baricentricCoordinates.z < distanceToClosestIntersection)) {
-            found = true;
-            distanceToClosestIntersection = baricentricCoordinates.z;
-            indexIntersectedPrimitive = i;
+        glm::vec3 unusedPos, unusedNormal;
+
+        // Speed optimization. Two pass hit test.
+        // First test if the ray hits the bounding sphere of the object
+        // and if it does, test if it really hits a triangle in the object.
+        // You can use this even if the shapes are not spheres.
+        // Testing against a bounding sphere is a very fast test.
+        // If it doesn't hit the bounding sphere then there
+        // is no point in testing against every single triangle in the mesh.
+
+        // Fast test
```

Comment:
Hello Abe. I prefer to move any other solution that deals with bounding boxes or any form of optimization of the mesh intersection algorithm in another example, this example is about mouse picker over a mesh. Also, I still have to think how to implement bounding boxes, or maybe a BVH tree. So, if you can remove this part, I am ready to merge.

Reply:
ok! Do you know if I can somehow update the PR? Or should I close it and make a new one?

Reply:
I guess I can delete those lines and push one more commit, that should work. Would that be ok?

Reply:
You can simply edit your branch locally, push your forked branch, and automatically your PR will be updated

Reply:
My question was if I should push one more time (so there would be 5 commits) removing the things from commit 4, or revert the last commit (so there would be 3 commits). I did the latter by running `git push --force`.